### PR TITLE
Added IfZeroVector Transformer

### PR DESCRIPTION
**Pull request by roshan-nair**
@akshay-sharma IfZeroVectorTransformer changes

### CODE REVIEW

File: adapters-1.6/src/main/java/com/flipkart/fdp/ml/adapter/IfZeroVectorModelInfoAdapter.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import com.flipkart.fdp.ml.IfZeroVector;
+import com.flipkart.fdp.ml.modelinfo.IfZeroVectorModelInfo;
+import org.apache.spark.sql.DataFrame;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+/**
+ * Transforms {@link IfZeroVector} in MlLib to  {@link IfZeroVectorModelInfo} object
```
Comment (akshay-sharma) [inline adapters-1.6/src/main/java/com/flipkart/fdp/ml/adapter/IfZeroVectorModelInfoAdapter.java]: remove mllib

File: custom-transformer/src/main/scala/com/flipkart/fdp/ml/IfZeroVector.scala [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  final def getThenSetValue: String = $(thenSetValue)
+
+  final val elseSetCol: Param[String] = new Param[String](this, "column", "a column name that exists in the dataframe")
+  final def getElseSetCol: String = $(elseSetCol)
+
+  /** Validates and transforms the input schema. */
+  protected def validateAndTransformSchema(schema: StructType): StructType = {
+    //val typeCandidates = List(new ArrayType(StringType, true), new ArrayType(StringType, false))
+    //CustomSchemaUtil.checkColumnTypes(schema, $(inputCol), typeCandidates)
+    CustomSchemaUtil.appendColumn(schema, $(outputCol), new VectorUDT)
```
Comment (akshay-sharma) [inline custom-transformer/src/main/scala/com/flipkart/fdp/ml/IfZeroVector.scala]: ouput column need not be a vector. this should be of the the type of  input col

File: custom-transformer/src/main/scala/com/flipkart/fdp/ml/IfZeroVector.scala [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+trait IfZeroVectorParams extends Params {
+  final val inputCol: Param[String] = new Param[String](this, "inputCol", "input column name")
+  final def getInputCol: String = $(inputCol)
+
+  final val outputCol: Param[String] = new Param[String](this, "outputCol", "output column name")
+  setDefault(outputCol, uid + "__output")
+  final def getOutputCol: String = $(outputCol)
+
+  final val thenSetValue: Param[String] = new Param[String](this, "string constant", "a string constant")
```
Comment (akshay-sharma) [inline custom-transformer/src/main/scala/com/flipkart/fdp/ml/IfZeroVector.scala]: this need not be a string constant

File: custom-transformer/src/main/scala/com/flipkart/fdp/ml/IfZeroVector.scala [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    dataFrame.withColumn($(outputCol), encode(col($(inputCol)), col($(elseSetCol))))
+  }
+
+  override def transformSchema(schema: StructType): StructType = {
+    val inputType = schema($(inputCol)).dataType
+    require(inputType.isInstanceOf[VectorUDT],
+      s"Input column ${$(inputCol)} must be a vector column")
+    require(!schema.fieldNames.contains($(outputCol)),
+      s"Output column ${$(outputCol)} already exists.")
+    return CustomSchemaUtil.appendColumn(schema, $(outputCol), new VectorUDT)
```
Comment (akshay-sharma) [inline custom-transformer/src/main/scala/com/flipkart/fdp/ml/IfZeroVector.scala]: outputCol should be of type setCol

File: custom-transformer/src/main/scala/com/flipkart/fdp/ml/IfZeroVector.scala (hunk 3 above)
Comment (roshan-nair) [inline custom-transformer/src/main/scala/com/flipkart/fdp/ml/IfZeroVector.scala]: Not doing this at the moment. We'll take this up when we plan to generalize this.

**Pull request merged**